numba/cuda/compiler.py

```diff
@@ -1,18 +1,21 @@
 from __future__ import absolute_import, print_function
 import copy
 import sys
+import warnings
+import operator
 
 from numba import ctypes_support as ctypes
 from numba.typing.templates import AbstractTemplate
 from numba import config, compiler, types
 from numba.typing.templates import ConcreteTemplate
-from numba import funcdesc, typing, utils
+from numba import funcdesc, typing, utils, serialize
 
 from .cudadrv.autotune import AutoTuner
 from .cudadrv.devices import get_context
 from .cudadrv import nvvm, devicearray, driver
 from .errors import KernelRuntimeError
 from .api import get_current_device
+from functools import reduce
```

Comment:
Hmm, just a detail but can you leave the order of imports consistent? :-) (first stdlib then third-party then package-wide then local imports).